src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,97 @@
+package codeu.controller;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Servlet class responsible for the profile page. */
+public class ProfileServlet extends HttpServlet {
+   
+  private MessageStore messageStore;
+
+  /** Store class that gives access to Users. */
+  private UserStore userStore;
+
+  /** Set up state for handling profile requests. */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setMessageStore(MessageStore.getInstance());
+    setUserStore(UserStore.getInstance());
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  /**
+   * This function fires when a user navigates to the profile page. It gets the profile name from
+   * the URL, finds the corresponding profile subject, and fetches the messages sent by that subject.
+   * It then forwards to profile.jsp for rendering.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    String requestUrl = request.getRequestURI();
+
+    if(requestUrl.length() <= "/profile/".length()){
+      // if user navigates to "/profile/" without a specific user
+      request.getRequestDispatcher("/index.jsp").forward(request,response);
+      return;
+    }
+    String subjectName = requestUrl.substring("/profile/".length());
+	
+    
+    User subject = userStore.getUser(subjectName);
+
+    if(subject == null) {
```

Comment:
Again, it's great you're covering this. For now, we can also redirect to '/' like I suggested above?
For later, how about showing an error page, like a 404 or something, when the user navigates to a page that doesn't exist? Just something to keep in mind.

Reply:
404 sounds like a good idea. Thanks!

Reply:
Cool. I wouldn't worry about it though, I'd open an issue and forget about it for now.